### ISSUE TITLE
Rename versionRange to availability

### ISSUE THREAD

**Issue by snicoll** [spring-io/initializr#418]
The `versionRange` attribute can be misleading since it is really about the range of Spring Boot versions that the dependency is compatible with. If we renamed the attribute to something like `availability` the intent would be more clear.

ping @dsyer

**Comment by dsyer** [comment]
Agree. I guess we need to retain the old name and deprecate it in parallel.

**Comment by shakuzen** [comment]
A bit verbose perhaps but something like `compatibleBootRange` might be a bit more clear in meaning.

**Comment by snicoll** [comment]
I am not keen to use `Boot` anywhere as a shortcut for Spring Boot. Ideally I'd like a single word, two words top.

**Comment by snicoll** [comment]
Brainstorming with @mbhave, we don't think `availability` is the right term as we lose the semantic of the fact it defines a "range". Perhaps there's a better term around "platform version"?

Flagging for team attention to see what the rest of the team thinks.

**Comment by wilkinsona** [comment]
How about `compatibilityRange`?

**Comment by snicoll** [comment]
Thanks Andy, let's go with `compatibilityRange `
